test/basicfuns.jl

```diff
@@ -44,6 +44,30 @@
     @test iszero(xlog1py(0 + im * 0, -1 + im * Inf))
 end
 
+@testset "xexpx" begin
+    for x in (0, 0.0, 0f0, -Inf)
+        @test iszero(@inferred xexpx(x))
+    end
+    @test isnan(@inferred xexpx(NaN))
+    @test Inf == @inferred xexpx(Inf)
+    @test exp(1) ≈ @inferred xexpx(1)
+    @test 2exp(2) ≈ @inferred xexpx(2)
+end
+
+@testset "xexpy" begin
+    @test iszero(@inferred xexpy(Inf, -Inf))
+    @test isnan(@inferred xexpy(NaN, -Inf))
+    @test isnan(@inferred xexpy(NaN, 1))
+    @test isnan(@inferred xexpy(0, NaN))
+    @test iszero(@inferred xexpy(0, -Inf))
+    @test ℯ ≈ @inferred xexpy(1, 1)
```

Comment:
Maybe rather use
```suggestion
    @test e ≈ @inferred xexpy(1, 1)
```
or
```suggestion
    @test exp(1) ≈ @inferred xexpy(1, 1)
```

Reply:
Ah wait, it's covered by the loop below already:
```suggestion
```

Reply:
That's right, thanks!